internal/connector/proxy.go

```diff
@@ -6,34 +6,49 @@ import (
 	"fmt"
 	"net/http"
 	"net/http/httputil"
+	"strconv"
 	"sync"
+	"time"
 
-	"github.com/gin-gonic/gin"
+	"github.com/prometheus/client_golang/prometheus"
 
 	"github.com/infrahq/infra/internal/certs"
 	"github.com/infrahq/infra/internal/logging"
+	"github.com/infrahq/infra/metrics"
 )
 
 func proxyMiddleware(
 	proxy *httputil.ReverseProxy,
 	authn *authenticator,
 	bearerToken string,
-) gin.HandlerFunc {
-	return func(c *gin.Context) {
-		claim, err := authn.Authenticate(c.Request)
+) func(resp http.ResponseWriter, req *http.Request) {
+	return func(resp http.ResponseWriter, req *http.Request) {
+		start := time.Now()
+		status := http.StatusOK
+		defer func() {
+			metrics.RequestDuration.With(prometheus.Labels{
+				"host":   req.Host,
+				"method": req.Method,
+				"path":   "proxy",
+				"status": strconv.Itoa(status),
```

Comment:
These two values changes slightly:

* previously we'd always report an empty string for `path`, because the proxy doesn't know anything about the route path (there is none). Change to use an explicit `proxy` which shows up better in datadog.
* `status` previously would report any status, but really all we care about in this case is if we rejected the request in the proxy or not. If the backend (kubernetes) happens to return a 4xx or 5xx response, that's not really relevant to this metric because it's not a failure of the proxy.